### ISSUE TITLE
[Feature Request] Spawn storage barricade with inventory (call spawn table)

### ISSUE THREAD

**Issue by obfuscateClarity** [SmartlyDressedGames/Unturned-3.x-Community#2415]
Hi! 
Working on a custom sentry mod, hoping for the ability to place a storage barricade with a pre-loaded inventory from a spawn table. 
Currently working around this using the carepackage script, however the drawback is that the sentry that spawns is hostile to me. 
Another use case is placeable ammo crates for PVP mods, etc. 
Thanks!

**Comment by obfuscateClarity** [comment]
(Alternatively, an option to spawn with a specific item ID would also be ok, i.e. spawn with item 99 to have a sentry spawn with a cobra in the inventory)

**Comment by obfuscateClarity** [comment]
Hey @SDGNelson sorry for the @ - just wondering if this one is still on your list as we are hoping to publish in about 1-2 weeks. Issues #2413 #2414 and #2415 are all related to the custom sentry mod. All our assets are complete and we have a proof of concept with the base sentry, just waiting on these features. Thanks again for your consideration! Cheers.

**Comment by SDGNelson** [comment]
Hmm while the suggestions do make sense, I do not have them on my to-do list. Did I say I would do them somewhere? If I said I would do them then I will of course honor my promise, but otherwise there are a lot of feature requests.

**Comment by obfuscateClarity** [comment]
> Hmm while the suggestions do make sense, I do not have them on my to-do list. Did I say I would do them somewhere? If I said I would do them then I will of course honor my promise, but otherwise there are a lot of feature requests.

Sorry @SDGNelson no you did not promise it anywhere, my misunderstanding. 
Use case here is the ability to spawn a sentry with a spotlight already in inventory, like a home alert system. 
Issue #2414 is so we can include custom alert sounds, like an air raid siren for example.
Issue #2413 is so we can track players and/or vehicles at a longer range, say like a fighter jet flying overhead at night. 

This would also allow for a sentry to be placed with a weapon in inventory, that would instantly protect the player. Right now this is achievable by using the care package script (i.e. calling spawn table) however when the barricade is placed, it is "un-owned" and becomes hostile to the player. So, you place a barricade, sentry spawns with a weapon, and it immediately shoots you. It's actually quite comical but contrary to our objective with the mod. 

If possible, please at the very least consider issue #2415 as has multiple use cases for us modders:
- Care packages that you can personally call and that are locked to you
- Crafting recipes that provide you with a final item inside a barricade you must place down
- Pre-armed sentries for faster deployment 
- This could even be used to create MAWS type "flare dispensers" for vehicles like helicopters and jets which dispense automatically when they detect a nearby enemy. 

A potential issue is item duplication through the salvaging of barricades - one simple solution is to make such barricades "unpickupable" in the .dat. 

Please consider issue #2415 at least, as this will have an immediate positive impact and allow for many more possibilities with barricades, items, and sentries. 

Thanks for your consideration Nelson!

**Comment by SDGNelson** [comment]
In the meantime an option might be to include a Spotlight component on the turret prefab's "Aim" transform.

On the surface this would be a straightforward addition, but because the savedata and netcode are so entangled it would be tricky to do as-is. More likely to happen after https://trello.com/c/Jmk1j7WU/32-persistence

**Comment by SDGNelson** [comment]
Yeah that is much more straightforward, will add Detection_Radius in the next update.

**Comment by obfuscateClarity** [comment]
@SDGNelson oh man I might reallyyy be pushing my luck here but if you're already looking at the sentries this week could you potentially also allow us to set our own <Acquired> and <Lost> effects instead of effect 92 and 93? 

Currently if a sentry is alerted, the effect manager calls effect 92. When the sentry loses lock, effect 93 is called.

Ideally there would be a .dat option for:

Effect_Acquired XXX
and
Effect_Lost XXX

or even 

**Target_Acquired XXX
Target_Lost XXX**

Thank you sorry for bugging you again!

**Comment by SDGNelson** [comment]
Yeah yeah okay I will mark it next update for the update after Kuwait.
